### PR TITLE
Update typescript version - Closes #6107

### LINKED ISSUE #6107: Update typescript version
### Description

Update typescript version and relevant development packages

### Motivation

Typescript has many new features and SDK should follow the latest version

### Acceptance Criteria

Build should pass

### PR DESCRIPTION
**Pull request by shuse2**
- Update typescript version
- Update @types/node version
- Update jest related packages
- Fix breaking changes introduced by typescript and jest packages

### What was the problem?

This PR resolves #6107 

### How was it solved?

- Update typescript and all related breaking changes

### How was it tested?

- All lint and build should pass

### CODE REVIEW

**Review by mitsuaki-u — APPROVED** [review]
LGTM

File: commander/test/bootstrapping/commands/passphrase/encrypt.spec.ts [hunk 1]
```diff
@@ -19,6 +19,10 @@ import * as readerUtils from '../../../../src/utils/reader';
 import { EncryptCommand } from '../../../../src/bootstrapping/commands/passphrase/encrypt';
 import { getConfig } from '../../../helpers/config';
 
+jest.mock('@liskhq/lisk-cryptography', () => ({
```
Comment (ManuGowda) [inline commander/test/bootstrapping/commands/passphrase/encrypt.spec.ts:22]: any specific reason now to mock the whole library? would be nice to know in what scenarios we need to mock the whole library due to typescript version change.

**Review by ManuGowda — APPROVED** [review]
LGTM.

File: commander/test/bootstrapping/commands/passphrase/encrypt.spec.ts (hunk 1 above)
Comment (shuse2) [reply commander/test/bootstrapping/commands/passphrase/encrypt.spec.ts:22]: it's not really mocking the whole library, but it's eliminating the getter. With new typescript version, the export is done as getter and cannot spy/mock on them unless replace like this

**Pull request merged**